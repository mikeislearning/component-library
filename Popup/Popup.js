import React from 'react';
import { Modal, SafeAreaView, ScrollView } from 'react-native';
import { PropTypes } from 'prop-types';
import HTML from 'react-native-render-html';

import { Button, EditFormHeader } from 'app/components';

import styles from './Popup.styles';

export default class Popup extends React.Component {
  static propTypes = {
    acceptButtonText: PropTypes.string,
    modalVisible: PropTypes.bool,
    onClose: PropTypes.func,
    openText: PropTypes.string,
    termsAccepted: PropTypes.bool,
    text: PropTypes.string,
  };

  static defaultProps = {
    modalVisible: false,
  };

  constructor(props) {
    super(props);

    this.state = {
      modalVisible: props.modalVisible,
    };
  }

  onClose = () => {
    this.setModalVisible(false);
    this.props.onClose();
  };

  setModalVisible = (visible) => {
    this.setState({ modalVisible: visible });
  };

  render() {
    const { openText, text } = this.props;
    if (!text) return null;

    return (
      <React.Fragment>
        <Modal animationType="slide" transparent={false} visible={this.state.modalVisible}>
          <SafeAreaView style={styles.safeAreaView}>
            <EditFormHeader onClose={this.onClose} />
            <ScrollView style={styles.container}>
              <HTML html={text} />
            </ScrollView>
          </SafeAreaView>
        </Modal>

        {openText && (
          <Button theme="quaternary" onPress={() => this.setModalVisible(true)}>
            {openText}
          </Button>
        )}
      </React.Fragment>
    );
  }
}
