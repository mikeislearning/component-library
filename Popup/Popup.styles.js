import { StyleSheet } from 'react-native';
import { colors } from 'app/theme';

const styles = StyleSheet.create({
  button: {
    alignSelf: 'center',
  },
  container: {
    flex: 1,
    padding: 20,
  },
  safeAreaView: {
    backgroundColor: colors.offwhite,
    flex: 1,
  },
});

export default styles;
