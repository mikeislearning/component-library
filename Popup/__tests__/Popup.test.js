/**
 * @format
 */

import React from 'react';
import { render } from 'react-native-testing-library';

import Popup from '../Popup';

describe('<Popup />', () => {
  it('renders correctly', () => {
    const popup = render(<Popup modalVisible={true} text="Content" />);
    expect(popup).toMatchSnapshot();
  });
});
