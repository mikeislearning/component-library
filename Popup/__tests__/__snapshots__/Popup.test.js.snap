// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Popup /> renders correctly 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  transparent={false}
  visible={true}
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "backgroundColor": "#FBFBFB",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "backgroundColor": "#FBFBFB",
          "borderBottomColor": "#ECECEC",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "height": 57,
          "justifyContent": "center",
          "paddingBottom": 12,
        }
      }
    >
      <View
        accessibilityLabel="Close"
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "bottom": 12,
            "left": 0,
            "opacity": 1,
            "paddingBottom": 5,
            "paddingLeft": 20,
            "position": "absolute",
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../app/images/close.png",
            }
          }
          style={
            Object {
              "height": 17,
              "width": 17,
            }
          }
        />
      </View>
    </View>
    <RCTScrollView
      style={
        Object {
          "flex": 1,
          "padding": 20,
        }
      }
    >
      <View>
        <View
          style={Object {}}
        >
          <Text
            allowFontScaling={true}
            selectable={false}
            style={
              Array [
                Object {},
                Object {},
              ]
            }
          >
            <Text
              allowFontScaling={true}
              style={
                Object {
                  "fontSize": 14,
                }
              }
            >
              Content
            </Text>
          </Text>
        </View>
      </View>
    </RCTScrollView>
  </RCTSafeAreaView>
</Modal>
`;
